src/Raven.Server/Documents/ShardedHandlers/ShardedQueriesHandler.cs

```diff
@@ -0,0 +1,744 @@
+                            //   other shards
+                            var includeTask = queryProcessor.HandleIncludes();
+                            if (includeTask.IsCompleted == false)
+                            {
+                                await includeTask.AsTask();
+                            }
+                            
+                            queryProcessor.MergeResults();
+                            
+                            // * For map/reduce - we need to re-run the reduce portion of the index again on the results
+                            queryProcessor.ReduceResults();
+
+                            // * For order by, we need to merge the results and compute the order again 
+                            queryProcessor.OrderResults();
+
+                            // * For map-reduce indexes we project the results after the reduce part 
+                            queryProcessor.ProjectAfterMapReduce();
+
+                            var result = queryProcessor.GetResult();
+                            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream(), HttpContext.RequestAborted))
+                            {
+                                result.Timings = indexQuery.Timings?.ToTimings();
+                                await writer.WriteDocumentQueryResultAsync(context, result, metadataOnly); 
+                                await writer.OuterFlushAsync();
+                            }
+
+
+                            // * For JS projections and load clauses, we don't support calling load() on a
+                            //   document that is not on the same shard
+                        }
+
+                    }
+                }
+                catch (Exception e)
+                {
+                    if (tracker.Query == null)
+                    {
+                        string errorMessage;
+                        if (e is EndOfStreamException || e is ArgumentException)
+                        {
+                            errorMessage = "Failed: " + e.Message;
+                        }
+                        else
+                        {
+                            errorMessage = "Failed: " +
+                                           HttpContext.Request.Path.Value +
+                                           e.ToString();
+                        }
+                        tracker.Query = errorMessage;
+                        if (TrafficWatchManager.HasRegisteredClients)
+                            AddStringToHttpContext(errorMessage, TrafficWatchChangeType.Queries);
+                    }
+                    throw;
+                }
+            }
+        }
+        
+        /// <summary>
+        /// TrafficWatchQuery writes query data to httpContext
+        /// </summary>
+        /// <param name="indexQuery"></param>
+        private void TrafficWatchQuery(IndexQueryServerSide indexQuery)
```

Comment:
this was copied from somewhere? avoid code duplication?

Reply:
done